src/orders/service/activities/OrderActivitiesService.ts

```diff
@@ -0,0 +1,90 @@
+import * as Q from 'q';
+import OrderActivity from './model/OrderActivity';
+import OrderActivityCreateRequestByChannel from './model/OrderActivityCreateRequestByChannel';
+import OrderActivityCreateRequestByOrderId from './model/OrderActivityCreateRequestByOrderId';
+import { AxiosRequestConfig, AxiosResponse } from 'axios';
+import RequestClientWrapper from '../../../RequestClientWrapper';
+import Resource from '../../../model/Resource';
+import Version from '../../../model/Version';
+import GenerateApiError from '../../../utils/GenerateApiError';
+import OrderActivityCreateRequestByBusiness
+  from './model/OrderActivityCreateRequestByBusiness';
+import OrdersCrudService from '../OrdersCrudService';
+
+const EXPECTED_CREATE_STATUS = 202;
+
+export default class OrdersActivitiesService {
+
+  constructor(
+    private readonly client: RequestClientWrapper,
+    private readonly ordersCrudService: OrdersCrudService
+  ) { }
+
+  public async create(orderActivityCreateRequest: OrderActivityCreateRequestByChannel): Promise<OrderActivity>;
+  public async create(orderActivityCreateRequest: OrderActivityCreateRequestByOrderId): Promise<OrderActivity>;
+  public async create(orderActivityCreateRequest: OrderActivityCreateRequestByBusiness): Promise<OrderActivity>;
+  public async create(
+    orderActivityCreateRequest:
+      OrderActivityCreateRequestByChannel &
+      OrderActivityCreateRequestByOrderId &
+      OrderActivityCreateRequestByBusiness
+  ): Promise<OrderActivity> {
+    const deferred = Q.defer<OrderActivity>();
+    if (orderActivityCreateRequest.businessId && orderActivityCreateRequest.channelOrderId) {
+      const order = await this.ordersCrudService.get({
+        businessId: orderActivityCreateRequest.businessId,
+        channelOrderId: orderActivityCreateRequest.channelOrderId
+      });
+      if (order.length === 0) {
+        deferred.reject(new Error(`Order could not be discerned,
+          no orders exist on businessId ${orderActivityCreateRequest.businessId} with channelOrderId of
+          ${orderActivityCreateRequest.channelOrderId}`
+        ));
+        return deferred.promise as any;
+      }
+      if (order.length > 1) {
+        deferred.reject(new Error(`Order could not be discerned,
+          ${order.length} orders exist on businessId ${orderActivityCreateRequest.businessId} with channelOrderId of
+          ${orderActivityCreateRequest.channelOrderId}`
+        ));
+        return deferred.promise as any;
+      }
+      orderActivityCreateRequest.orderId = order[0].id;
+    }
+    const requestUrl = `${Version.V1}${Resource.ORDERS_ACTIVITY}`;
+    const options: AxiosRequestConfig = {
+      data: orderActivityCreateRequest
+    };
+    if (orderActivityCreateRequest.actionId) {
```

Comment:
This is incorrect. The order activities endpoint takes the actionId in the request body which you are already passing. What we need you to do is pass X-Channel-Ape-Action-Id as a header for `/v1/orders` POST and PUT

Create order:
https://docs.channelape.io/#794dc5a2-7220-47e3-adf8-5e60f565ccc8

Update order:
https://docs.channelape.io/#24c35266-2906-4d95-a186-5ee35958ac88

Create order activity:
https://docs.channelape.io/#65e8a5c2-7077-4b6a-aaf2-d4799277b1d6